diff -ru a/usr/lib/linuxmint/mintUpdate/mintUpdate.py b/usr/lib/linuxmint/mintUpdate/mintUpdate.py
--- a/usr/lib/linuxmint/mintUpdate/mintUpdate.py	2022-10-30 16:37:07.467364670 +0300
+++ b/usr/lib/linuxmint/mintUpdate/mintUpdate.py	2022-10-30 16:30:58.142913940 +0300
@@ -810,24 +810,7 @@
             # Check presence of Mint layer
             (mint_layer_found, error_msg) = self.check_policy()
             if os.getenv("MINTUPDATE_TEST") == "layer-error" or (not mint_layer_found):
-                error_found = True
-                self.application.logger.write_error("Error: The APT policy is incorrect!")
-
-                label1 = _("Your APT configuration is corrupt.")
-                label2 = _("Do not install or update anything, it could break your operating system!")
-                label3 = _("To switch to a different Linux Mint mirror and solve this problem, click OK.")
-
-                msg = _("Your APT configuration is corrupt.")
-                if error_msg:
-                    error_msg = "\n\n%s\n%s" % (_("APT error:"), error_msg)
-                else:
-                    error_msg = ""
-                self.application.show_infobar(_("Please switch to another Linux Mint mirror"),
-                    msg, Gtk.MessageType.ERROR,
-                    callback=self._on_infobar_mintsources_response)
-                self.application.set_status(_("Could not refresh the list of updates"),
-                    "%s\n%s" % (label1, label2), "mintupdate-error-symbolic", True)
-                self.application.builder.get_object("label_error_details").set_markup("<b>%s\n%s\n%s%s</b>" % (label1, label2, label3, error_msg))
+                error_found = False
 
             if error_found:
                 Gdk.threads_enter()
Only in b/usr/lib/linuxmint/mintUpdate: __pycache__
diff -ru a/usr/share/linuxmint/mintupdate/main.ui b/usr/share/linuxmint/mintupdate/main.ui
--- a/usr/share/linuxmint/mintupdate/main.ui	2022-10-30 16:37:07.499365206 +0300
+++ b/usr/share/linuxmint/mintupdate/main.ui	2022-10-30 16:35:40.813811292 +0300
@@ -788,7 +788,7 @@
                       <object class="GtkLabel" id="label_welcome10">
                         <property name="visible">True</property>
                         <property name="can_focus">False</property>
-                        <property name="label" translatable="yes">Linux Mint recommends to set up automated system snapshots and to apply all available updates. This way, your computer is always secure and you can go back in time and recover from any potential regression.</property>
+                        <property name="label" translatable="yes">PikaOS recommends to set up automated system snapshots and to apply all available updates. This way, your computer is always secure and you can go back in time and recover from any potential regression.</property>
                         <property name="wrap">True</property>
                       </object>
                       <packing>
diff -ru a/etc/xdg/autostart/mintupdate.desktop b/etc/xdg/autostart/mintupdate.desktop
--- a/etc/xdg/autostart/mintupdate.desktop	2022-10-30 17:03:41.000000000 +0300
+++ b/etc/xdg/autostart/mintupdate.desktop	2022-11-03 21:19:20.027251849 +0300
@@ -148,7 +148,7 @@
 Name[zh_CN]=更新管理器
 Name[zh_HK]=更新管理員
 Name[zh_TW]=更新管理員
-Comment=Linux Mint Update Manager
+Comment=PikaOS Update Manager
 Icon=mintupdate
 Exec=mintupdate-launcher
 Terminal=false
