From: "Matteo F. Vescovi" <mfv@debian.org>
Date: Sun, 4 Sep 2016 14:32:42 +0200
Subject: Fix_FTBFS_on_kFreeBSD

---
 config/Makefile.kfreebsd | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

Index: glew-2.2.0/config/Makefile.kfreebsd
===================================================================
--- glew-2.2.0.orig/config/Makefile.kfreebsd
+++ glew-2.2.0/config/Makefile.kfreebsd
@@ -1,7 +1,14 @@
 NAME = $(GLEW_NAME)
 CC = cc
 LD = cc
-LDFLAGS.EXTRA = -L/usr/X11R6/lib
+M_ARCH ?= $(shell uname -m)
+ifeq (x86_64,${M_ARCH})
+  LDFLAGS.EXTRA = -L/usr/X11R6/lib64 -L/usr/lib64
+  LIBDIR = $(GLEW_DEST)/lib64
+else
+  LDFLAGS.EXTRA = -L/usr/X11R6/lib -L/usr/lib
+  LIBDIR = $(GLEW_DEST)/lib
+endif
 LDFLAGS.GL = -lGL -lX11
 LDFLAGS.STATIC = -Wl,-Bstatic
 LDFLAGS.DYNAMIC = -Wl,-Bdynamic
@@ -14,4 +21,4 @@ LIB.SONAME    = lib$(NAME).so.$(SO_MAJOR
 LIB.DEVLNK    = lib$(NAME).so
 LIB.SHARED    = lib$(NAME).so.$(SO_VERSION)
 LIB.STATIC    = lib$(NAME).a
-LDFLAGS.SO    = -shared -Wl,-soname $(LIB.SONAME)
+LDFLAGS.SO    = -shared -Wl,-soname=$(LIB.SONAME)
